### DISCUSSION TITLE
Idempotency?

### DISCUSSION THREAD

**Discussion by bluearcher-bc** [saltstack/salt-ext-modules-vmware#180] [Q&A]
Are these modules intended to be idempotent? Specifically doing vmware_cluster.create on a cluster name that already exists throws expected errors in vCenter and from salt saying the cluster already exists. I have testing suitability to use salt as sort of a desired state manager for configuration drift on clusters and hosts within vCenter.

I am new to Salt, so perhaps I am projecting too many expectations from a history with Ansible?

**Comment by waynew** [comment]
Hey @BlueArcherX  thanks for the question! Just saw this - apparently my Inbox was a little noisy :disappointed: 

Anyway - Salt /modules/ aren't idempotent, states are. We're currently building out the stateful functionality, so if you're missing some states that you're interested in, definitely open an issue for that!

Poking around at the code, it looks like that stateful functionality isn't something we have yet.
